### PR TITLE
Baseops

### PR DESCRIPTION
**Pull request by luitjens**
Added shape/stride to base ops and added lvalue to reverse/shift

### CODE REVIEW

File: include/matx_tensor_impl.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ class BaseOp
    * @param ev CUDA event
    * @param stream CUDA stream
    */
-  void run(cudaEvent_t ev, cudaStream_t stream = 0) noexcept
+  inline void run(cudaEvent_t ev, cudaStream_t stream = 0) noexcept
```
Comment (cliffburdick) [inline include/matx_tensor_impl.h:77]: `__MATX_INLINE__`

File: include/matx_tensor_impl.h [hunk 2]
```diff
@@ -87,11 +87,32 @@ class BaseOp
    * @param ex Executor
    */
   template <typename Ex>
-  void run (Ex ex) {
+  inline void run (Ex ex) {
```
Comment (cliffburdick) [inline include/matx_tensor_impl.h:90]: `__MATX_INLINE__`

File: include/matx_tensor_impl.h [hunk 3]
```diff
@@ -87,11 +87,32 @@ class BaseOp
...
     static_assert(is_executor_t<Ex>(), "Ex must be a MatX executor type");
     exec(*static_cast<T *>(this), ex);
   }
 
+  inline __MATX_HOST__ __MATX_DEVICE__ auto Shape() {
```
Comment (cliffburdick) [inline include/matx_tensor_impl.h:95]: `__MATX_INLINE__`

File: include/matx_tensor_ops.h [hunk 4]
```diff
@@ -616,11 +616,8 @@ inline
       else 
         return op_.Size(dim);
     }
-    
-    template<typename R> 
-    inline auto operator=(const R &rhs) {
-      return set(*this, rhs);
-    }
+  
+    template<typename R> inline auto operator=(const R &rhs) { return set(*this, rhs); }
```
Comment (cliffburdick) [inline include/matx_tensor_ops.h:620]: Was this supposed to be different?

Comment (luitjens) [reply include/matx_tensor_ops.h:620]: nope just consistent

**Pull request merged**